src/main/java/io/jenkins/pluginhealth/scoring/probes/LastCommitDateProbe.java

```diff
@@ -0,0 +1,85 @@
+package io.jenkins.pluginhealth.scoring.probes;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.ZonedDateTime;
+import java.util.Comparator;
+import java.util.stream.Stream;
+
+import io.jenkins.pluginhealth.scoring.model.Plugin;
+import io.jenkins.pluginhealth.scoring.model.ProbeResult;
+import io.jenkins.pluginhealth.scoring.model.ResultStatus;
+
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.eclipse.jgit.lib.Constants;
+import org.eclipse.jgit.lib.ObjectId;
+import org.eclipse.jgit.revwalk.RevCommit;
+import org.eclipse.jgit.revwalk.RevWalk;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.core.annotation.Order;
+import org.springframework.stereotype.Component;
+
+/**
+ * Using the analysis done by {@link SCMLinkValidationProbe}, this probe determines the last commit date of the plugin's repository.
+ */
+@Component
+@Order(value = LastCommitDateProbe.ORDER)
+public class LastCommitDateProbe extends Probe {
+    private static final Logger LOGGER = LoggerFactory.getLogger(LastCommitDateProbe.class);
+
+    public static final int ORDER = SCMLinkValidationProbe.ORDER + 1;
+    public Path tempDirectory;
+
+    @Override
+    public ProbeResult doApply(Plugin plugin) {
+        if (plugin.getDetails().get(SCMLinkValidationProbe.KEY) == null) {
+            LOGGER.error("Couldn't run {} on {} because previous SCMLinkValidationProbe has null value in database", key(), plugin.getName());
+            return ProbeResult.error(key(), "SCM link has not been probed yet");
+        }
+
+        if (plugin.getDetails().get(SCMLinkValidationProbe.KEY).status() == ResultStatus.SUCCESS) {
+            try {
+                tempDirectory = Files.createTempDirectory(plugin.getName());
+                try (Git git = Git.cloneRepository().setURI(plugin.getScm()).setDirectory(tempDirectory.toFile()).call()) {
+                    final ObjectId head = git.getRepository().resolve(Constants.HEAD);
+                    final RevCommit commit = new RevWalk(git.getRepository()).parseCommit(head);
+                    final ZonedDateTime zonedDateTime = ZonedDateTime.ofInstant(
+                        commit.getAuthorIdent().getWhenAsInstant(),
+                        commit.getAuthorIdent().getZoneId()
+                    );
+                    return ProbeResult.success(key(), zonedDateTime.toString());
+                } catch (GitAPIException ex)  {
```

Comment:
We'll come into this catch block when there's a problem in the `call()` method above, while we try to clone the plugin repo.

I ran this on DB and it is quite helpful because now I see that there are some plugins whose status for `SCMLinkValidationProbe` is a SUCCESS, but for this probe, the status is FAILURE with a message `"Could not clone the plugin repository"`.

For example, there are some plugins (https://github.com/jenkinsci/aws-java-sdk-plugin/aws-java-sdk-logs) with valid SCM as per the GH_REGEXP but we get 404 (as per my understanding, this plugin should fail SCMLinkValidationProbe too) on its SCM link so this probe fails to clone it.

Reply:
technically, the SCM is valid because the plugin is a submodule of the repository. But it is true that it is not a valid link to clone the repository.

Reply:
that is returning a `404`? `SCMLinkValidationProbe`? But still have a `success`?

Reply:
![image](https://user-images.githubusercontent.com/43273420/187872855-e3be486c-92f5-4227-bae1-6034fbac92e8.png)
I see the status code as `404` under the "Status" tab on the bottom right. I thought that's the same status code we're checking via `response.statusCode()`

Reply:
That is normal. The `SCMLinkValidationProbe` is not taking into account submodules to validate the repository. This means our code coverage is not enough. Working on it.

Reply:
> technically, the SCM is valid because the plugin is a submodule of the repository.

So now I understand what you mean by "plugin is a submodule of the repository". If I visit this link: https://github.com/jenkinsci/aws-java-sdk-plugin/ I can see there are "submodules" like `-ec2`, `-cs`, `-efs` etc. And these submodules are a plugin on their own (I might be wrong here) that's why they are distributed separately in the update-center.

> But it is true that it is not a valid link to clone the repository.

So, I guess there's no separate SCM Link to clone each of these submodules because that doesn't make sense as all of them reside in a single repo.

Reply:
> So now I understand what you mean by "plugin is a submodule of the repository".

it is a Maven module, not a Git submodule (which also is a existing concept). Sorry if I made things more complex to understand.

> And these submodules are a plugin on their own (I might be wrong here) that's why they are distributed separately in the update-center.

Correct.

> So, I guess there's no separate SCM Link to clone each of these submodules because that doesn't make sense as all of them reside in a single repo.

There is, but I'm not sure JGit can clone only a part of the repository. I'm investigating.